plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -140,26 +135,26 @@ public CacheStatsMBean getMetadataCacheStats()
     }
 
     public TableSnapshot loadSnapshot(SchemaTableName table, Path tableLocation, ConnectorSession session)
-            throws IOException
+            throws IOException, ExecutionException
```

Comment:
I think it would be better to unwrap the `ExecutionException`.
(also, watch out for `UncheckedExecutionException` and handle the same way)